.travis.yml

```diff
@@ -107,11 +107,15 @@ jobs:
       services:
         - docker
       before_install:
-        - docker run -dit --name emscripten -v $(pwd):/src trzeci/emscripten:sdk-incoming-64bit bash
+        # TODO: Emscripten upstream produces a broken build in CI environments only :(
```

Comment:
Is this fixing an actual issue on this CI? (If so why don't tests fail?)

If it's trying to prevent an issue we only see on other CI, is that necessary?

Reply:
This is a common thing on Travis, which cancels jobs when it doesn't see console output for 10 minutes, usually affecting long-running build tasks like this one (also see comment above). They have a bash function named `travis_wait` for this, but the last time I checked it suppressed output while the task executes so one can't watch the build.

Reply:
Just noticed that your question was about something different. Tests [do indeed fail](https://travis-ci.org/WebAssembly/binaryen/jobs/639244336#L471) when using the `emscripten-upstream` image, but it works when using the legacy `emscripten-fastcomp` image. This is happening since the first day Emscripten switched to upstream by default and I started testing it, so I thought a TODO would be appropriate to raise awareness. This is a very odd problem in that

* It compiles fine, but the output is broken (hits assertions)
* It only happens on CI (both Travis and GitHub Actions)
* I can't reproduce it locally
* It does not affect debug builds
* It affects any release build (without closure etc. etc.)
* Obviously isolated to building Binaryen.js

Reply:
This sounds worth investigating.   The fact that you can't repro it locally is very strange, especially since using the docker image should produce identical results.

I guess I'm OK landing this if its urgent but I'd rather try to fix instead.

Reply:
This PR is unrelated to the upstream breakage, though, it only updates the outdated fastcomp image to a recent fastcomp image (both work) and adds the TODO comment.

Reply:
Oh I see `sdk-incoming` is also fastcomp?

Reply:
Yeah, I believe that's Emscripten before upstream became the default, and they then switched to the other naming.